src/csharp/Microsoft.Spark.Worker/Processor/TaskContextProcessor.cs

```diff
@@ -22,29 +22,40 @@ internal TaskContext Process(Stream stream)
             return (_version.Major, _version.Minor) switch
             {
                 (2, 4) => TaskContextProcessorV2_4_X.Process(stream),
-                (3, _) => TaskContextProcessorV3_0_X.Process(stream),
+                (3, _) t when t.Minor < 4 => TaskContextProcessorV3_0_X.Process(stream),
+                (3, _) => TaskContextProcessorV3_5_X.Process(stream),
                 _ => throw new NotSupportedException($"Spark {_version} not supported.")
             };
         }
 
-        private static TaskContext ReadTaskContext(Stream stream)
+        private static TaskContext ReadTaskContext_2_x(Stream stream)
+        => new()
         {
-            return new TaskContext
-            {
-                StageId = SerDe.ReadInt32(stream),
-                PartitionId = SerDe.ReadInt32(stream),
-                AttemptNumber = SerDe.ReadInt32(stream),
-                AttemptId = SerDe.ReadInt64(stream)
-            };
-        }
+            IsBarrier = SerDe.ReadBool(stream),
+            Port = SerDe.ReadInt32(stream),
+            Secret = SerDe.ReadString(stream),
+
+            StageId = SerDe.ReadInt32(stream),
+            PartitionId = SerDe.ReadInt32(stream),
+            AttemptNumber = SerDe.ReadInt32(stream),
+            AttemptId = SerDe.ReadInt64(stream),
+        };
 
-        private static void ReadBarrierInfo(Stream stream)
+        // Needed for 3.3.4+, 3.4.x, 3.5.x
```

Comment:
Could you please double check the version? From the jira: https://issues.apache.org/jira/browse/SPARK-36173, the cpus field is added from spark 3.3.0.
![1732702995520](https://github.com/user-attachments/assets/61cdbea7-aaca-47b4-95dc-9442f271e1b4)

Reply:
Fixed, thanks for attentiveness